### PR TITLE
feat: print flexible nowdoc

### LINKED ISSUE #854: Support php7.3 and php7.4
No issue template. We need improve parser (https://github.com/glayzzle/php-parser):

# php7.3

- [x] Support trailing commas in `call`/`isset`/`unset` (spec https://wiki.php.net/rfc/trailing-comma-function-calls)
- [ ] Support new syntax for NowDoc and HereDoc (spec https://wiki.php.net/rfc/flexible_heredoc_nowdoc_syntaxes)
- [x] Array Destructuring supports Reference Assignments
- [x] Instanceof Operator accepts Literals (i think we already support, need add tests)

# php7.4

- [x] https://wiki.php.net/rfc/null_coalesce_equal_operator
- [x] https://wiki.php.net/rfc/typed_properties_v2
- [x] https://wiki.php.net/rfc/arrow_functions_v2
- [x] https://wiki.php.net/rfc/spread_operator_for_array
- [x] https://wiki.php.net/rfc/numeric_literal_separator

Something missing? Any help welcome :+1:

### LINKED ISSUE #1292: Nowdoc: invalid trailing comma
```php
call(
    $var
        ? "foo"
        : <<<'EOD'
        string
        string
        string
        EOD
);
```
is formatted as
```php
call(
    $var
        ? "foo"
        : <<<'EOD'
        string
        string
        string
        EOD,
);
```
which is invalid.

### PR DESCRIPTION
**Pull request by czosel**
Fixes #1292
Fixed #854 

Turns out to be quite tricky to implement the rule suggested in https://github.com/prettier/plugin-php/issues/1274#issuecomment-570709723 - this is a simpler version.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ Add the following configuration to `.vscode/launch.json`:
         "<node_internals>/**"
     ],
     "program": "${workspaceRoot}/node_modules/.bin/prettier",
-    "args": ["--plugin=.", "--parser=php", "foo.php"],
```
Comment (czosel) [inline CONTRIBUTING.md]: drive-by docs fix

Comment (loilo) [reply CONTRIBUTING.md]: If you say so. 😄

**Comment by czosel** [comment]
I took another look, and unfortunately couldn't figure out how to determine if the parent group breaks and indent accordingly. I removed the indentation for now, since this is already an improvement over the current printing.

@evilebottnawi Do you have an idea how to implement this? https://github.com/prettier/plugin-php/issues/1274#issuecomment-570709723

**Comment by czosel** [comment]
Tests are failing because of #1292

**Comment by czosel** [comment]
Tests are passing, i'll update the PR once https://github.com/glayzzle/php-parser/pull/509 has landed :rocket:

**Comment by czosel** [comment]
The parser bug has been fixed! I think we have everything ready for our `0.14.0` release featuring support for prettier 2.0 and for PHP 7.3 and 7.4! Would be awesome if someone could have a look at the open merge requests :upside_down_face: 
@evilebottnawi @loilo
